### DISCUSSION TITLE
How to set javac options?

### DISCUSSION THREAD

**Discussion by kubukoz** [VirtusLab/scala-cli#1301] [General]
In sbt (for example), one can compile their Scala to a specific version of JVM bytecode by specifying `javacOptions`:

```scala
javacOptions ++= Seq("-source", "8", "-target", "8")
```

I was unable to do that in scala-cli. I couldn't find docs on how to do it via directives, but found some about CLI parameters, so I tried:

```bash
scala-cli package .  -f --javac-option -source --javac-option 1.8 --javac-option -target --javac-option 1.8
```

with this app:

```scala
//> using scala "3.1.3"

object Main {
  def main(args: Array[String]) = args.zipWithIndex.foreach(println)
}
```

My JVM (Zulu openjdk11) didn't like this:

```bash
./Main

Unrecognized option: -source
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

What am I doing wrong? Is there a directive that'd allow me to pass all these flags inside the source file?

**Comment by lwronski** [comment]
@kubukoz 
Thanks for reporting, it seems there is some regression in scala-cli, I created issue to fix this bug - https://github.com/VirtusLab/scala-cli/issues/1326

If you use scala-cli `0.1.9` it should works:
```
scala-cli --cli-version 0.1.9 package .  -f --javac-option -source --javac-option 1.8 --javac-option -target --javac-option 1.8
```

**Reply by kubukoz** [reply]
How would you configure these options via directives?

**Reply by lwronski** [reply]
@kubukoz Sorry for the long answer, you can try to use `javacOpt` using the directive that was added in the last release of ` 0.1.16` - https://scala-cli.virtuslab.org/docs/commands/compile/#jvm-options.
